lib/Runtime/Library/ForInObjectEnumerator.cpp

```diff
@@ -182,7 +207,32 @@ namespace Js
                 if (TestAndSetEnumerated(propertyId) //checks if the property is already enumerated or not
                     && (attributes & PropertyEnumerable))
                 {
-                    return currentIndex;
+                    bool propertyShadowed = false;
```

Comment:
Usually almost all the properties of built-in prototype are non-enumerable properties, and skip them for look-up is a good acceleration, like we could ignore visiting ~20 methods on Array.prototype which save us time. Array.prototype and Object.prototype share no  properties so bitmap doesn't help this much.  Also lookup in descendant prototype is guarded by bitmap test, so there would be minimum extra look-up introduced.
I will run benchmark to see whether there is some significant perf impact.